src/agent/commands/actions.js

```diff
@@ -233,5 +235,23 @@ export const actionsList = [
             await agent.npc.setGoal(name, quantity);
             return 'Set goal: ' + agent.npc.data.curr_goal.name;
         }
+    },
+    {
+        name: '!selfPrompt',
+        description: 'Continously prompt yourself to continue acting without user input.',
+        params: {
+            'prompt': '(string) The starting prompt.',
+        },
+        perform: async function (agent, prompt) {
+            agent.self_prompter.start(prompt); // don't await, don't return
+        }
+    },
+    {
+        name: '!stopSelfPrompt',
+        description: 'Stop current action and self-prompting.',
+        perform: async function (agent) {
+            agent.self_prompter.stop();
+            return 'Self-prompting stopped.';
+        }
```

Comment:
well, yes, but this command is a special case, where !stopSelfPrompt can be called by the bot as a response to a user message, but not by itself when it's self-prompting so it will keep running endlessly until you ask it to stop.

we probably should make a user_callable_only flag but I say leave it for a future pr